libclc/generic/lib/integer/ctz.cl

```diff
@@ -0,0 +1,42 @@
+#include "../clcmacro.h"
+#include <clc/clc.h>
+
+_CLC_OVERLOAD _CLC_DEF short ctz(short x) {
+  return x ? __builtin_ctzs(x) : 16;
+}
+
+_CLC_OVERLOAD _CLC_DEF ushort ctz(ushort x) {
+  return x ? __builtin_ctzs(x) : 16;
+}
+
+_CLC_OVERLOAD _CLC_DEF char ctz(char x) {
+  return x ? ctz((ushort)(uchar)x) : 8;
+}
+_CLC_OVERLOAD _CLC_DEF uchar ctz(uchar x) {
+  return x ? ctz((ushort)x) : 8;
+}
+
+_CLC_OVERLOAD _CLC_DEF int ctz(int x) {
+  return x ? __builtin_ctz(x) : 32;
+}
+
+_CLC_OVERLOAD _CLC_DEF uint ctz(uint ux) {
+  return ux ? __builtin_ctz(ux) : 32;
```

Comment:
Consistent with llvm way. It is recommended to change "ux" to "x", like "uchar" uses "x".